{"dependencies": [{"name": "aiohappyeyeballs", "version": "2.6.1", "vulns": []}, {"name": "aiohttp", "version": "3.12.14", "vulns": []}, {"name": "aiosignal", "version": "1.4.0", "vulns": []}, {"name": "aiosqlite", "version": "0.21.0", "vulns": []}, {"name": "anaconda-anon-usage", "skip_reason": "Dependency not found on PyPI and could not be audited: anaconda-anon-usage (0.7.1)"}, {"name": "annotated-types", "version": "0.6.0", "vulns": []}, {"name": "anyio", "version": "4.9.0", "vulns": []}, {"name": "archspec", "version": "0.2.3", "vulns": []}, {"name": "attrs", "version": "25.3.0", "vulns": []}, {"name": "banks", "version": "2.1.3", "vulns": []}, {"name": "beautifulsoup4", "version": "4.13.4", "vulns": []}, {"name": "boltons", "version": "24.1.0", "vulns": []}, {"name": "boolean-py", "version": "5.0", "vulns": []}, {"name": "brotli", "version": "1.0.9", "vulns": []}, {"name": "cachecontrol", "version": "0.14.3", "vulns": []}, {"name": "cachetools", "version": "5.5.2", "vulns": []}, {"name": "certifi", "version": "2025.4.26", "vulns": []}, {"name": "cffi", "version": "1.17.1", "vulns": []}, {"name": "charset-normalizer", "version": "3.3.2", "vulns": []}, {"name": "click", "version": "8.2.1", "vulns": []}, {"name": "colorama", "version": "0.4.6", "vulns": []}, {"name": "conda", "skip_reason": "Dependency not found on PyPI and could not be audited: conda (25.5.1)"}, {"name": "conda-anaconda-telemetry", "skip_reason": "Dependency not found on PyPI and could not be audited: conda-anaconda-telemetry (0.1.2)"}, {"name": "conda-anaconda-tos", "skip_reason": "Dependency not found on PyPI and could not be audited: conda-anaconda-tos (0.2.0)"}, {"name": "conda-content-trust", "skip_reason": "Dependency not found on PyPI and could not be audited: conda-content-trust (0.2.0)"}, {"name": "conda-libmamba-solver", "skip_reason": "Dependency not found on PyPI and could not be audited: conda-libmamba-solver (25.4.0)"}, {"name": "conda-package-handling", "version": "2.4.0", "vulns": []}, {"name": "conda-package-streaming", "version": "0.11.0", "vulns": []}, {"name": "contourpy", "version": "1.3.2", "vulns": []}, {"name": "coverage", "version": "7.9.2", "vulns": []}, {"name": "cryptography", "version": "45.0.3", "vulns": []}, {"name": "cycler", "version": "0.12.1", "vulns": []}, {"name": "cyclonedx-python-lib", "version": "9.1.0", "vulns": []}, {"name": "dataclasses-json", "version": "0.6.7", "vulns": []}, {"name": "defusedxml", "version": "0.7.1", "vulns": []}, {"name": "deprecated", "version": "1.2.18", "vulns": []}, {"name": "dirtyjson", "version": "1.0.8", "vulns": []}, {"name": "distro", "version": "1.9.0", "vulns": []}, {"name": "execnet", "version": "2.1.1", "vulns": []}, {"name": "fastapi", "version": "0.116.1", "vulns": []}, {"name": "filelock", "version": "3.18.0", "vulns": []}, {"name": "filetype", "version": "1.2.0", "vulns": []}, {"name": "fonttools", "version": "4.58.5", "vulns": []}, {"name": "frozendict", "version": "2.4.2", "vulns": []}, {"name": "frozenlist", "version": "1.7.0", "vulns": []}, {"name": "fsspec", "version": "2025.5.1", "vulns": []}, {"name": "google-ai-generativelanguage", "version": "0.6.15", "vulns": []}, {"name": "google-api-core", "version": "2.25.1", "vulns": []}, {"name": "google-api-python-client", "version": "2.176.0", "vulns": []}, {"name": "google-auth", "version": "2.40.3", "vulns": []}, {"name": "google-auth-httplib2", "version": "0.2.0", "vulns": []}, {"name": "google-genai", "version": "1.25.0", "vulns": []}, {"name": "google-generativeai", "version": "0.8.5", "vulns": []}, {"name": "googleapis-common-protos", "version": "1.70.0", "vulns": []}, {"name": "greenlet", "version": "3.2.3", "vulns": []}, {"name": "griffe", "version": "1.7.3", "vulns": []}, {"name": "grpcio", "version": "1.73.1", "vulns": []}, {"name": "grpcio-status", "version": "1.71.2", "vulns": []}, {"name": "h11", "version": "0.16.0", "vulns": []}, {"name": "httpcore", "version": "1.0.9", "vulns": []}, {"name": "httplib2", "version": "0.22.0", "vulns": []}, {"name": "httptools", "version": "0.6.4", "vulns": []}, {"name": "httpx", "version": "0.28.1", "vulns": []}, {"name": "idna", "version": "3.7", "vulns": []}, {"name": "iniconfig", "version": "2.1.0", "vulns": []}, {"name": "jinja2", "version": "3.1.6", "vulns": []}, {"name": "jiter", "version": "0.10.0", "vulns": []}, {"name": "joblib", "version": "1.5.1", "vulns": []}, {"name": "jsonpatch", "version": "1.33", "vulns": []}, {"name": "jsonpointer", "version": "2.1", "vulns": []}, {"name": "kiwisolver", "version": "1.4.8", "vulns": []}, {"name": "libmambapy", "skip_reason": "Dependency not found on PyPI and could not be audited: libmambapy (2.0.5)"}, {"name": "license-expression", "version": "30.4.3", "vulns": []}, {"name": "llama-cloud", "version": "0.1.32", "vulns": []}, {"name": "llama-cloud-services", "version": "0.6.43", "vulns": []}, {"name": "llama-index", "version": "0.12.45", "vulns": []}, {"name": "llama-index-agent-openai", "version": "0.4.12", "vulns": []}, {"name": "llama-index-cli", "version": "0.4.4", "vulns": []}, {"name": "llama-index-core", "version": "0.12.48", "vulns": []}, {"name": "llama-index-embeddings-google-genai", "version": "0.2.1", "vulns": []}, {"name": "llama-index-embeddings-openai", "version": "0.3.1", "vulns": []}, {"name": "llama-index-indices-managed-llama-cloud", "version": "0.7.10", "vulns": []}, {"name": "llama-index-instrumentation", "version": "0.2.0", "vulns": []}, {"name": "llama-index-llms-google-genai", "version": "0.2.4", "vulns": []}, {"name": "llama-index-llms-openai", "version": "0.4.7", "vulns": []}, {"name": "llama-index-multi-modal-llms-openai", "version": "0.5.3", "vulns": []}, {"name": "llama-index-program-openai", "version": "0.3.2", "vulns": []}, {"name": "llama-index-question-gen-openai", "version": "0.3.1", "vulns": []}, {"name": "llama-index-readers-file", "version": "0.4.11", "vulns": []}, {"name": "llama-index-readers-llama-parse", "version": "0.4.0", "vulns": []}, {"name": "llama-index-workflows", "version": "1.1.0", "vulns": []}, {"name": "llama-parse", "version": "0.6.43", "vulns": []}, {"name": "markdown-it-py", "version": "2.2.0", "vulns": []}, {"name": "markupsafe", "version": "3.0.2", "vulns": []}, {"name": "marshmallow", "version": "3.26.1", "vulns": []}, {"name": "matplotlib", "version": "3.10.3", "vulns": []}, {"name": "mdurl", "version": "0.1.0", "vulns": []}, {"name": "menuinst", "skip_reason": "Dependency not found on PyPI and could not be audited: menuinst (2.2.0)"}, {"name": "mpmath", "version": "1.3.0", "vulns": []}, {"name": "msgpack", "version": "1.1.1", "vulns": []}, {"name": "multidict", "version": "6.6.3", "vulns": []}, {"name": "mypy-extensions", "version": "1.1.0", "vulns": []}, {"name": "nest-asyncio", "version": "1.6.0", "vulns": []}, {"name": "networkx", "version": "3.5", "vulns": []}, {"name": "nltk", "version": "3.9.1", "vulns": []}, {"name": "numpy", "version": "2.3.1", "vulns": []}, {"name": "openai", "version": "1.95.1", "vulns": []}, {"name": "packageurl-python", "version": "0.17.1", "vulns": []}, {"name": "packaging", "version": "24.2", "vulns": []}, {"name": "pandas", "version": "2.2.3", "vulns": []}, {"name": "pillow", "version": "11.3.0", "vulns": []}, {"name": "pip", "version": "25.1", "vulns": []}, {"name": "pip-api", "version": "0.0.34", "vulns": []}, {"name": "pip-audit", "version": "2.9.0", "vulns": []}, {"name": "pip-requirements-parser", "version": "32.0.1", "vulns": []}, {"name": "platformdirs", "version": "4.3.7", "vulns": []}, {"name": "playwright", "version": "1.53.0", "vulns": []}, {"name": "pluggy", "version": "1.5.0", "vulns": []}, {"name": "propcache", "version": "0.3.2", "vulns": []}, {"name": "proto-plus", "version": "1.26.1", "vulns": []}, {"name": "protobuf", "version": "5.29.5", "vulns": []}, {"name": "psutil", "version": "7.0.0", "vulns": []}, {"name": "py-serializable", "version": "2.0.0", "vulns": []}, {"name": "pyasn1", "version": "0.6.1", "vulns": []}, {"name": "pyasn1-modules", "version": "0.4.2", "vulns": []}, {"name": "pycosat", "version": "0.6.6", "vulns": []}, {"name": "pycparser", "version": "2.21", "vulns": []}, {"name": "pydantic", "version": "2.11.7", "vulns": []}, {"name": "pydantic-core", "version": "2.33.2", "vulns": []}, {"name": "pyee", "version": "13.0.0", "vulns": []}, {"name": "pygments", "version": "2.19.1", "vulns": []}, {"name": "pymupdf", "version": "1.26.3", "vulns": []}, {"name": "pyparsing", "version": "3.2.3", "vulns": []}, {"name": "pypdf", "version": "5.8.0", "vulns": []}, {"name": "pysocks", "version": "1.7.1", "vulns": []}, {"name": "pytest", "version": "8.4.1", "vulns": []}, {"name": "pytest-base-url", "version": "2.1.0", "vulns": []}, {"name": "pytest-cov", "version": "6.2.1", "vulns": []}, {"name": "pytest-html", "version": "4.1.1", "vulns": []}, {"name": "pytest-metadata", "version": "3.1.1", "vulns": []}, {"name": "pytest-mock", "version": "3.14.1", "vulns": []}, {"name": "pytest-playwright", "version": "0.7.0", "vulns": []}, {"name": "pytest-xdist", "version": "3.8.0", "vulns": []}, {"name": "python-dateutil", "version": "2.9.0.post0", "vulns": []}, {"name": "python-dotenv", "version": "1.1.1", "vulns": []}, {"name": "python-multipart", "version": "0.0.20", "vulns": []}, {"name": "python-slugify", "version": "8.0.4", "vulns": []}, {"name": "pytz", "version": "2025.2", "vulns": []}, {"name": "pyyaml", "version": "6.0.2", "vulns": []}, {"name": "regex", "version": "2024.11.6", "vulns": []}, {"name": "requests", "version": "2.32.3", "vulns": [{"id": "GHSA-9hjg-9r4m-mvj7", "fix_versions": ["2.32.4"], "aliases": ["CVE-2024-47081"], "description": "### Impact  Due to a URL parsing issue, Requests releases prior to 2.32.4 may leak .netrc credentials to third parties for specific maliciously-crafted URLs.  ### Workarounds For older versions of Requests, use of the .netrc file can be disabled with `trust_env=False` on your Requests Session ([docs](https://requests.readthedocs.io/en/latest/api/#requests.Session.trust_env)).  ### References https://github.com/psf/requests/pull/6965 https://seclists.org/fulldisclosure/2025/Jun/2"}]}, {"name": "rich", "version": "13.9.4", "vulns": []}, {"name": "rsa", "version": "4.9.1", "vulns": []}, {"name": "ruamel-yaml", "version": "0.18.10", "vulns": []}, {"name": "ruamel-yaml-clib", "version": "0.2.12", "vulns": []}, {"name": "seaborn", "version": "0.13.2", "vulns": []}, {"name": "setuptools", "version": "80.9.0", "vulns": []}, {"name": "six", "version": "1.17.0", "vulns": []}, {"name": "sniffio", "version": "1.3.1", "vulns": []}, {"name": "sortedcontainers", "version": "2.4.0", "vulns": []}, {"name": "soupsieve", "version": "2.7", "vulns": []}, {"name": "sqlalchemy", "version": "2.0.41", "vulns": []}, {"name": "starlette", "version": "0.47.1", "vulns": []}, {"name": "striprtf", "version": "0.0.26", "vulns": []}, {"name": "sympy", "version": "1.14.0", "vulns": []}, {"name": "tenacity", "version": "8.5.0", "vulns": []}, {"name": "text-unidecode", "version": "1.3", "vulns": []}, {"name": "tiktoken", "version": "0.9.0", "vulns": []}, {"name": "toml", "version": "0.10.2", "vulns": []}, {"name": "torch", "version": "2.7.1", "vulns": [{"id": "GHSA-887c-mr87-cxwp", "fix_versions": [], "aliases": ["CVE-2025-3730"], "description": "A vulnerability, which was classified as problematic, was found in PyTorch 2.6.0. Affected is the function torch.nn.functional.ctc_loss of the file aten/src/ATen/native/LossCTC.cpp. The manipulation leads to denial of service. An attack has to be approached locally. The exploit has been disclosed to the public and may be used. The name of the patch is 46fc5d8e360127361211cb237d5f9eef0223e567. It is recommended to apply a patch to fix this issue."}]}, {"name": "tqdm", "version": "4.67.1", "vulns": []}, {"name": "truststore", "version": "0.10.0", "vulns": []}, {"name": "typing-extensions", "version": "4.12.2", "vulns": []}, {"name": "typing-inspect", "version": "0.9.0", "vulns": []}, {"name": "typing-inspection", "version": "0.4.1", "vulns": []}, {"name": "tzdata", "version": "2025.2", "vulns": []}, {"name": "uritemplate", "version": "4.2.0", "vulns": []}, {"name": "urllib3", "version": "2.3.0", "vulns": [{"id": "GHSA-48p4-8xcf-vxj5", "fix_versions": ["2.5.0"], "aliases": ["CVE-2025-50182"], "description": "urllib3 [supports](https://urllib3.readthedocs.io/en/2.4.0/reference/contrib/emscripten.html) being used in a Pyodide runtime utilizing the [JavaScript Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) or falling back on [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest). This means you can use Python libraries to make HTTP requests from your browser or Node.js. Additionally, urllib3 provides [a mechanism](https://urllib3.readthedocs.io/en/2.4.0/user-guide.html#retrying-requests) to control redirects.  However, the `retries` and `redirect` parameters are ignored with Pyodide; the runtime itself determines redirect behavior.   ## Affected usages  Any code which relies on urllib3 to control the number of redirects for an HTTP request in a Pyodide runtime.   ## Impact  Redirects are often used to exploit SSRF vulnerabilities. An application attempting to mitigate SSRF or open redirect vulnerabilities by disabling redirects may remain vulnerable if a Pyodide runtime redirect mechanism is unsuitable.   ## Remediation  If you use urllib3 in Node.js, upgrade to a patched version of urllib3.  Unfortunately, browsers provide no suitable way which urllib3 can use: `XMLHttpRequest` provides no control over redirects, the Fetch API returns `opaqueredirect` responses lacking data when redirects are controlled manually. Expect default browser behavior for redirects."}, {"id": "GHSA-pq67-6m6q-mj2v", "fix_versions": ["2.5.0"], "aliases": ["CVE-2025-50181"], "description": "urllib3 handles redirects and retries using the same mechanism, which is controlled by the `Retry` object. The most common way to disable redirects is at the request level, as follows:  ```python resp = urllib3.request(\"GET\", \"https://httpbin.org/redirect/1\", redirect=False) print(resp.status) # 302 ```  However, it is also possible to disable redirects, for all requests, by instantiating a `PoolManager` and specifying `retries` in a way that disable redirects:  ```python import urllib3  http = urllib3.PoolManager(retries=0)  # should raise MaxRetryError on redirect http = urllib3.PoolManager(retries=urllib3.Retry(redirect=0))  # equivalent to the above http = urllib3.PoolManager(retries=False)  # should return the first response  resp = http.request(\"GET\", \"https://httpbin.org/redirect/1\") ```  However, the `retries` parameter is currently ignored, which means all the above examples don't disable redirects.  ## Affected usages  Passing `retries` on `PoolManager` instantiation to disable redirects or restrict their number.  By default, requests and botocore users are not affected.  ## Impact  Redirects are often used to exploit SSRF vulnerabilities. An application attempting to mitigate SSRF or open redirect vulnerabilities by disabling redirects at the PoolManager level will remain vulnerable.  ## Remediation  You can remediate this vulnerability with the following steps:   * Upgrade to a patched version of urllib3. If your organization would benefit from the continued support of urllib3 1.x, please contact [sethmichaellarson@gmail.com](mailto:sethmichaellarson@gmail.com) to discuss sponsorship or contribution opportunities.  * Disable redirects at the `request()` level instead of the `PoolManager()` level."}]}, {"name": "uvicorn", "version": "0.35.0", "vulns": []}, {"name": "watchfiles", "version": "1.1.0", "vulns": []}, {"name": "websockets", "version": "15.0.1", "vulns": []}, {"name": "wheel", "version": "0.45.1", "vulns": []}, {"name": "win-inet-pton", "version": "1.1.0", "vulns": []}, {"name": "wrapt", "version": "1.17.2", "vulns": []}, {"name": "yarl", "version": "1.20.1", "vulns": []}, {"name": "zstandard", "version": "0.23.0", "vulns": []}], "fixes": []}
